### PR TITLE
pull: Authenticate using instructions in WWW-Authenticate

### PR DESCRIPTION
**Pull request by reidpr**
Address #655 and #737.

### CODE REVIEW

**Comment by trandles-lanl** [comment]
Works for me from my yellow linux workstation. I tried from fg-fey and it didn't work but that's because HTTP access to git.lanl.gov isn't enabled, only ssh access works. Is it possible to pull an image via ssh?

**Comment by reidpr** [comment]
I thought push/pull was an HTTP(S) only protocol? I could be wrong though.

File: lib/charliecloud.py [hunk 1]
```diff
@@ -597,47 +587,145 @@ def from_tree(self, t):
...
+         fp.write(res.content)
+         fp.close()
+      except OSError as x:
+         FATAL("can't write: %s: %s" % (path, x))
+
+   def get_layer(self, hash_, path):
+      "GET the layer with hash hash_ and save it at path."
+      # /v1/library/hello-world/blobs/<layer-hash>
+      url = self._url_of("blobs", "sha256:" + hash_)
+      accept = "application/vnd.docker.image.rootfs.diff.tar.gzip"
```
Comment (j-ogas) [inline lib/charliecloud.py:697]: If we don't already, we should include a URL where this media type stems from.

Comment (reidpr) [reply lib/charliecloud.py:697]: Turns out we do, on line 208 in `Image.download()`.

**Pull request merged**